### PR TITLE
PLAT-126321: Add UI tests for ContextualPopupDecorator

### PR DESCRIPTION
**Pull request by adrian-cocoara-lgp**
Enact-DCO-1.0-Signed-off-by: Adrian Cocoara adrian.cocoara@lgepartner.com

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
 - Added ui tests for ContextualPopupDecorator
 - Updated unit tests

### Links
[//]: # (Related issues, references)
PLAT-126321

### Comments

### CODE REVIEW

**Review by stanca-pop-lgp — APPROVED** [review]

File: tests/ui/specs/ContextualPopupDecorator/ContextualPopupDecoratorPage.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+	hover () {
+		return $(this.selector).moveTo({xOffset: 0, yOffset: 0});
+	}
+
+	get self () {
+		return $(this.selector);
+	}
+
+	get isButtonExist () {
```
Comment (daniel-stoian-lgp) [inline tests/ui/specs/ContextualPopupDecorator/ContextualPopupDecoratorPage.js:22]: do you use this?

File: tests/ui/specs/ContextualPopupDecorator/ContextualPopupDecoratorPage.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	constructor (id) {
+		this.id = id;
+		this.selector = `#${this.id}`;
+	}
+
+	focus () {
+		return browser.execute((el) => el.focus(), $(this.selector));
+	}
+
+	hover () {
```
Comment (daniel-stoian-lgp) [inline tests/ui/specs/ContextualPopupDecorator/ContextualPopupDecoratorPage.js:14]: this one also

File: tests/ui/specs/ContextualPopupDecorator/ContextualPopupDecoratorPage.js (hunk 1 above)
Comment (adrian-cocoara-lgp) [reply tests/ui/specs/ContextualPopupDecorator/ContextualPopupDecoratorPage.js:22]: Oops. I forgot to remove them, thanks

**Review by daniel-stoian-lgp — APPROVED** [review]

**Review by jeonghee27 — APPROVED** [review]
LGTM

File: tests/ui/apps/ContextualPopupDecorator/ContextualPopupDecorator-View.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	<Button id="popupButton1">Hello Contextual Popup1</Button>
+);
+const renderPopup2 = () => (
+	<Button id="popupButton2">Hello Contextual Popup2</Button>
+);
+
+class app extends Component {
+	state = {
+		button1State: false,
+		button2State: false
```
Comment (MikyungKim) [inline tests/ui/apps/ContextualPopupDecorator/ContextualPopupDecorator-View.js:23]: How about renaming these as `button#Open` ?

File: tests/ui/specs/ContextualPopupDecorator/ContextualPopupDecorator-specs.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+	describe('not using open', function () {
+		it('should focus the first button on start', function () {
+			expect(button1.self.isFocused()).to.be.true();
+		});
+
+		describe('using 5-way', function () {
+			// The menu is open and the first item has focus
+			beforeEach(function () {
+				Page.spotlightSelect();
```
Comment (MikyungKim) [inline tests/ui/specs/ContextualPopupDecorator/ContextualPopupDecorator-specs.js:21]: Shouldn't this be only for the first `it`?

File: tests/ui/apps/ContextualPopupDecorator/ContextualPopupDecorator-View.js (hunk 3 above)
Comment (adrian-cocoara-lgp) [reply tests/ui/apps/ContextualPopupDecorator/ContextualPopupDecorator-View.js:23]: Sure.

File: tests/ui/specs/ContextualPopupDecorator/ContextualPopupDecorator-specs.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+		button1,
+		button2
+	} = Page.components;
+
+	describe('not using open', function () {
+		it('should focus the first button on start', function () {
+			expect(button1.self.isFocused()).to.be.true();
+		});
+
+		describe('using 5-way', function () {
```
Comment (MikyungKim) [inline tests/ui/specs/ContextualPopupDecorator/ContextualPopupDecorator-specs.js:18]: How about adding 'using pointer' case?
And also closing the contextual popup by clicking?

Comment (adrian-cocoara-lgp) [reply tests/ui/specs/ContextualPopupDecorator/ContextualPopupDecorator-specs.js:18]: Thanks for suggestion! I added some more tests.

**Review by MikyungKim — APPROVED** [review]
LGTM

**Pull request merged**